verifier/testreport.py

```diff
@@ -219,17 +246,231 @@ def createHtmlDiffReport(self):
+            'html_file_name': os.path.join(exec, html_name)
+        }
+
+      except BaseException as err:
+        print('SUMMARIZE REPORTS for file %f. Error:  %s' % (f, err))
+
+      try:
+        # Categorize by exec and test_type
+        if exec not in self.exec_summary:
+          self.exec_summary[exec] = [test_results]
+        else:
+          self.exec_summary[exec].append(test_results)
+
+        if test_type not in self.type_summary:
+          self.type_summary[test_type] = [test_results]
+        else:
+          self.type_summary[test_type].append(test_results)
+
+      except BaseException as err:
+        print('SUMMARIZE REPORTS in exec_summary %s, %s. Error: %s' % (exec, test_type, err))
+
+    if self.debug > 0:
+      #print('SUMMARY exec: %s' % self.exec_summary)
+      print('SUMMARY type: %s' % self.type_summary)
 
-    # Generate HTML page containing this information
     return
 
+  def getStats(self, entry):
+    # Process items in a map to give HTML table value
+    outList = []
+    outList.append('Test count: %s' % entry['test_count'])
+    outList.append('Succeeded: %s' % entry['pass_count'])
+    outList.append('Failed: %s' % entry['fail_count'])
+    outList.append('Errors: %s' % entry['error_count'])
+    outList.append('Missing verify: %s' % entry['missing_verify_count'])
+    outList.append('<a href="%s"  target="_blank">Details</a>' %
+                   entry['html_file_name'])
+    return '<br>'.join(outList) + '</a>'
+
+  def createSummaryHtml(self):
+    # Generate HTML page containing this information
+    # Create the template
+    html_map = {
+        'all_platforms': ', '.join(list(self.exec_summary.keys())),
+        'all_tests': ', '.join(list(self.type_summary.keys())),
+        'datetime':  datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
+    }
+    # Create header for each executor
+    header_line = ''  # TODO
+    html_map['header_line'] = header_line
+
+    # Create row for each test
+    detail_lines = []
+    html_map['detail_lines'] = detail_lines
+
+    # For each test in sorted(self.type_summary):
+    #   For each exec in test:
+    exec_set = set()
+    for exec in self.exec_summary:
+      exec_set.add(exec)
+
+    exec_list = sorted(list(exec_set))
+    # Get table header
+    header_list = ['<th>Test type</th>']
+    for exec in exec_list:
+      header_vals = {'header_data': exec}
+      header_list.append(self.headerItemTemplate.safe_substitute(header_vals))
+
+    header_line = self.lineTemplate.safe_substitute(
+        { 'column_data': ''.join(header_list) }
+    )
+    html_map['exec_header_line'] = header_line
+    print('&&&&&& header_line = %s' % (header_line))
+
+    if self.debug > 1:
+      print('EXEC SET = %s' % exec_list)
+      print('Header = %s' % header_line)
+
+    # Generate a row containing the test data, including the test type
+    # and results for each test in its column
+    data_rows = []
+    row_items = []
+    for test in sorted(self.type_summary):
```

Comment:
We should use a templating language for this in the near future. Consider making a standalone PR to switch this over to a templating language (but not before we finish the rest of the quarterly goals)

Reply:
Yes, I'll add a TODO for including a templating language for these reports.